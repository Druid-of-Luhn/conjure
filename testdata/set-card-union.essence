language Essence 2.0

find x,y : set (maxSize 4) of int(0..9)

such that
    $ card(x union y) = 2,
    x != y
