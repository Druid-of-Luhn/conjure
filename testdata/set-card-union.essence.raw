Spec
  { language = "Essence"
  , version = [ 2
              , 0
              ]
  , topLevelBindings = [ ( Find
                         , "x"
                         , DomainSet
                             { size = Nothing
                             , minSize = Nothing
                             , maxSize = Just ( ValueInteger 4 )
                             , attrDontCare = False
                             , element = DomainIntegerFromTo (
                                           Just ( ValueInteger 0 ) ) (
                                           Just ( ValueInteger 9 ) )
                             , representation = Nothing
                             }
                         )
                       , ( Find
                         , "y"
                         , DomainSet
                             { size = Nothing
                             , minSize = Nothing
                             , maxSize = Just ( ValueInteger 4 )
                             , attrDontCare = False
                             , element = DomainIntegerFromTo (
                                           Just ( ValueInteger 0 ) ) (
                                           Just ( ValueInteger 9 ) )
                             , representation = Nothing
                             }
                         )
                       ]
  , topLevelWheres = []
  , objective = Nothing
  , constraints = [ GenericNode
                      Neq
                      [ Identifier "x"
                      , Identifier "y"
                      ]
                  ]
  , metadata = []
  }