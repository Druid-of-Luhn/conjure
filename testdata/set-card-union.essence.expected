language Essence 2.0

find x : set (maxSize 4) of int(0..9)
find y : set (maxSize 4) of int(0..9)

such that
    x != y
